certora/specs/AccessControlDefaultAdminRules.spec

```diff
@@ -338,8 +348,8 @@ rule acceptDefaultAdminTransfer(env e) {
 */
 rule cancelDefaultAdminTransfer(env e) {
   require nonpayable(e);
-  requireInvariant defaultAdminConsistency(defaultAdmin());
-  requireInvariant singleDefaultAdmin(e.msg.sender, defaultAdmin());
+  require nonzerosender(e);
```

Comment:
Why is this needed? Even if the zero address is the sender and the `defaultAdmin()`, I think it's something theoretically allowed

Reply:
Lets imagine both `msg.sender` and the `defaultAdmin()` are 0. We are in a case where the admin was renounced. In that case, even if `defaultAdmin()` is 0, `hasRole(DEFAULT_ADMIN_ROLE, address(0))` is false.

So the call to `changeDefaultAdminDelay` will fail because `onlyRole(DEFAULT_ADMIN_ROLE)` will not pass. This causes the liveness check to fail.

Either we do 
```
require nonzerosender(e)
```
or we change the liveness check to
```
assert success <=> (e.msg.sender == defaultAdmin() && e.msg.sender != 0),
    "only the current default admin can cancel a transfer";
```

I think the first option is simpler

Reply:
Got it, thanks for the explanation!